### ISSUE TITLE
On-canvas UI isn't visible until after first app click

### ISSUE THREAD

**Issue by iwehrman** [adobe-photoshop/spaces-design#2316]
1. Create a new document.
2. Reload Design Space from remote Dev Tools
3. Cmd-tab back to Design space.
4. Hover cursor over a guide zone.

Result: 
The guide zone doesn't show. Once you click either the canvas or the HTML UI, it starts working. 

Sometimes I also see the guide zone appear when the cursor is in the middle of the canvas, but I can't immediately figure out repro steps for that.

====

Update: This seems to affect all on-canvas UI, including the border overlays. (The policy frames and UI bounds seem to be correct though.)

**Comment by iwehrman** [comment]
Reported with Mac pgdev.563 at d90d3fe84047c87233c10af4292fed13a252e631.

**Comment by iwehrman** [comment]
I also note that updating the window transform (which happens, for example, when you pan the canvas) does not help. I'm also definitely seeing the thing where the UI shows up at unexpected times w.r.t. my cursor position regularly, but I still can't figure out concrete repro steps.

**Comment by iwehrman** [comment]
I now think the problem is strictly that we're getting bad mousemove events from PS until the first click. I can reliably trigger the guide zone by moving my mouse around the middle of canvas, though sometimes it takes some hunting to find the right place. Also, interestingly, I can never repro this when I try to use QuickTime to make a screen recording of the problem...

**Comment by iwehrman** [comment]
@jfitz-adobe @chadrolfs can you repro this? @volfied was unable to do so.

**Comment by jfitz-adobe** [comment]
Can't repro using Spaces v1.0.35 + HTML Sprint30.
Both Mac & Win seem to work OK.

**Comment by volfied** [comment]
@iwehrman if you can repro it, see if you can get the transformation matrix when this is happening. All those drawings rely on transformation matrix being correct, so maybe our initial values are messing things up and we should not draw anything when transform matrix is not available.

**Comment by chadrolfs** [comment]
Here's another easy way to repro:
View > Show Guides (uncheck)
View > Show Guides (check)

Now you're in the state where guide zone will either not show or flicker until you click on the canvas.

**Comment by volfied** [comment]
Well I just yoinked this, but I'll be fixing the issue where re-showing guides etc. does not re-draw the overlays.

**Comment by volfied** [comment]
I de-yoink it, this has to do with HTML/PS focus battle more than anything.

**Comment by iwehrman** [comment]
I'm still seeing this. I don't know what the issue is. What's the focus issue here, @volfied?

**Comment by iwehrman** [comment]
I can still repro this 100% reliably. @volfied: can you clarify your focus comment above?

**Comment by volfied** [comment]
I think the problem causing this was the fact that OS.externalMouseMove events were sending us wrong coordinates after using menu items (or focusing back through dev tools like Ian does).

This is the externalMouseMove locations we get after clicking the menu item:
```
[418.8515625, -438]
 [418.8515625, -438] 
// At this point, I click on the canvas, bringing focus back to PS.
 [415.8515625, 106] -> Correct coordinates
 [415.8515625, 106]
```

**Comment by chadrolfs** [comment]
https://watsonexp.corp.adobe.com/#bug=4068004

**Comment by chadrolfs** [comment]
(from watson bug) Jesper - 
On OSX simply clicking in the menu cause this problem to happen.

On OSX: This is caused by a confusion between the target window for OS events when we get a hitTest. In a number of cases the window is nil, so GetViewRelativeLocation is iffy. We may have to switch to obtaining global coordinates and then converting to source views in os_event

**Comment by chadrolfs** [comment]
deferrable IMO

**Comment by jsbache** [comment]
Fixed in staging in CL: 1048497.
I was not able to see a problem on Windows and the fix is OSX only. Let me know if you are seeing an issue on Windows.
Root of the problem is that [NSEvent locationInWindow] sometimes returns global coordinates.
